[FIX] web_editor: fix html field content not saved first time

jvm-odoo opened the pull request:
### Issue

- Install Email Marketing
- Create/Edit one of them
- Save

Changes not saved

- Click edit & save

Changes saved (sometimes)

### Cause

Introduced in 30db768
Very weird, it seems to be a promises order issue

### Solution

Do the save before saveImages

**OPW-2207107
OPW-2207817**

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

jvm-odoo commented:
@nle-odoo

jvm-odoo commented:
Runbot red, I will check for another solution

jvm-odoo commented:
@nle-odoo I think runbot will be green this time

jvm-odoo commented:
I don't why it worked because what you say make sense. However, I found the real issue.

Situation: you create a new mass mailing and save the content which is "test"

Before 30db768, the order of execution was:
1. field_html > commitChanges (value = <p><br></p>)
2. mass_mailing_widget > commitChanges (value = "test")

After 30db768:
1. mass_mailing_widget > commitChanges (value = "test")
2. field_html > commitChanges (value = <p><br></p>)

I guess this is because the field_html now take more time to be done.

I don't know if we really want to do field_html.commitChanges when using mass_mailing_widget because we do the same + some new processing in mass_mailing_widget.commitChanges.

I will push a new solution which fix the issue by forcing field_html.commitChanges to be done before mass_mailing_widget.commitChanges and removing the duplicated processing.

nle-odoo commented:
Looks good, this is big so makes me afraid.

If the issue is just the latency added by `this.wysiwyg.saveCroppedImages` (and solved by calling the super that will do it too), maybe we can just also add it in the MassMailingFieldHtml `commitChanges` directly.

jvm-odoo commented:
The diff looks big but I just added
```javascript
 this._super.apply(this, arguments).then(function () {
```

and removed the 2 first `if`

I don't think that adding `this.wysiwyg.saveCroppedImages` is a good idea because if in the future we add another promise in `field_html` the issue will repeat

nle-odoo commented:
you can add `?w=1` to see diff ignoring whitespace: https://github.com/odoo/odoo/pull/46864/files?w=1

> I don't think that adding this.wysiwyg.saveCroppedImages

but then:

- we do 2 times self.wysiwyg.save and 2 times ` self._isDirty = result.isDirty`

- we call `super` that we never did before

but maybe that's not an issue

jvm-odoo commented:
Should be good!
